jxl/src/frame.rs

```diff
@@ -604,181 +609,133 @@ impl Frame {
 
 #[cfg(test)]
 mod test {
-    use std::{panic, path::Path};
-
-    use jxl_macros::for_each_test_file;
+    use std::panic;
 
     use crate::{
-        bit_reader::BitReader,
         container::ContainerParser,
+        decode::{decode_jxl_codestream, DecodeOptions},
         error::Error,
         features::spline::Point,
-        headers::{FileHeader, JxlHeader},
-        icc::read_icc,
         util::test::assert_almost_eq,
     };
     use test_log::test;
 
-    use super::{DecoderState, Frame, Section};
+    use super::Frame;
 
+    #[allow(clippy::type_complexity)]
     fn read_frames(
         image: &[u8],
-        mut callback: impl FnMut(Frame) -> Result<Option<DecoderState>, Error>,
+        callback: Option<&mut dyn FnMut(&Frame) -> Result<(), Error>>,
```

Comment:
probably should not have the Option here :-)

Reply:
Done.